### PR TITLE
Refactor: Move Dockerfile to a separate docker directory

### PR DESCRIPTION
**Pull request by Vyom-Yadav**
# Summary

***Provide a brief overview of the changes and the issue being addressed. 
Explain the rationale and any background necessary for understanding the changes. 
List dependencies required by this change, if any.***

The reminder service would also have a Dockerfile, so moving the Dockerfile from the root to a dedicated folder. Make commands, `docker compose` commands will continue to work as usual.

## Change Type

***Mark the type of change your PR introduces:***

- [ ] Bug fix (resolves an issue without affecting existing features)
- [ ] Feature (adds new functionality without breaking changes)
- [ ] Breaking change (may impact existing functionalities or require documentation updates)
- [x] Documentation (updates or additions to documentation)
- [x] Refactoring or test improvements (no bug fixes or new functionality)

# Testing

***Outline how the changes were tested, including steps to reproduce and any relevant configurations. 
Attach screenshots if helpful.***

# Review Checklist:

- [x] Reviewed my own code for quality and clarity.
- [ ] Added comments to complex or tricky code sections.
- [ ] Updated any affected documentation.
- [ ] Included tests that validate the fix or feature.
- [ ] Checked that related changes are merged.

### CODE REVIEW

File: .mk/develop.mk [hunk 1]
```diff
@@ -42,7 +47,7 @@ run-docker: run-docker-teardown ## run the app under docker-compose
 	@# podman (at least) doesn't seem to like multi-arch images, and sometimes picks the wrong one (e.g. amd64 on arm64)
 	@# We also need to remove the build: directives to use ko builds
 	@# ko resolve will fill in the image: field in the compose file, but it adds a yaml document separator
-	@sed -e '/^  *build:/d'  -e 's|  image: minder:latest|  image: ko://github.com/stacklok/minder/cmd/server|' docker-compose.yaml | KO_DOCKER_REPO=$(KO_DOCKER_REPO) ko resolve --base-import-paths --platform linux/$(DOCKERARCH) -f - | sed 's/^--*$$//' > .resolved-compose.yaml
+	@yq e $(YQ_BUILD_REPLACE_STRING) docker-compose.yaml | KO_DOCKER_REPO=$(KO_DOCKER_REPO) ko resolve --base-import-paths --platform linux/$(DOCKERARCH) -f - | sed 's/^--*$$//' > .resolved-compose.yaml
```
Comment (jhrozek) [inline .mk/develop.mk:50]: Could we do without `yq` or fix the docs (README, docs/docs/developer_guide/get-hacking.md and docs/docs/run_minder_server/run_the_server.md) to include yq installation?

Comment (Vyom-Yadav) [reply .mk/develop.mk:50]: > Could we do without yq

We can, but `yq` is becoming pretty standard. Even GH-hosted runners have it installed. 

> fix the docs (README, docs/docs/developer_guide/get-hacking.md and docs/docs/run_minder_server/run_the_server.md) to include yq installation

Yeah, I will add it to docs.

Comment (jhrozek) [reply .mk/develop.mk:50]: another thing might be to put it into the make bootstrap command in `.mk/develop.mk`

Comment (Vyom-Yadav) [reply .mk/develop.mk:50]: Done, added docs and to the `bootstrap` command.

Comment (jhrozek) [reply .mk/develop.mk:50]: thank you. It seems that you'd need to install `github.com/mikefarah/yq/v4@latest`, without the latest tag I get:

```bash
go install github.com/mikefarah/yq/v4
no required module provides package github.com/mikefarah/yq/v4; to add it:
        go get github.com/mikefarah/yq/v4
```

Comment (Vyom-Yadav) [reply .mk/develop.mk:50]: `cd tools && go mod tidy`? It should work after that

**Comment by jhrozek** [comment]
@Vyom-Yadav sorry, can you resolve the conflict? I'll approve right afterwards

**Comment by Vyom-Yadav** [comment]
Done @jhrozek

**Review by jhrozek — APPROVED** [review]

**Pull request merged**